src/App.tsx

```diff
@@ -15,47 +18,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export function getSortedGoods(
+  goods: string[],
+  state: State,
```

Comment:
I don't think that passing state as props is a good idea here, because after adding other props to the state - we will pass the args we don't need. 

What I meant with my comment about avoiding too many props in the func is that you should separate functions, but not pass props as objects. Feel free to ask me if you have further questions about this comment

Reply:
will ask on QnA